examples/tcp_client/tcp_client.cc

```diff
@@ -0,0 +1,157 @@
+/**************************************************************************/ /**
+ * @brief Linux TCP client example.
+ * @file
+ ******************************************************************************/
+
+#include <cerrno>
+#include <cmath> // For lround()
+#include <csignal> // For signal()
+#include <cstdio> // For fprintf()
+#include <cstring> // For memcpy()
+#include <string> // For stoi() and strerror()
+
+#include <netdb.h> // For gethostbyname() and hostent
+#include <netinet/in.h> // For IPPROTO_* macros and htons()
+#include <sys/socket.h> // For socket support.
+#include <unistd.h> // For close()
+
+#include <point_one/fusion_engine/messages/core.h>
+#include <point_one/fusion_engine/parsers/fusion_engine_framer.h>
+
+using namespace point_one::fusion_engine::messages;
+using namespace point_one::fusion_engine::parsers;
+
+static bool shutdown_pending_ = false;
+
+/******************************************************************************/
+void HandleSignal(int signal) {
+  if (signal == SIGINT || signal == SIGTERM) {
+    std::signal(signal, SIG_DFL);
+    shutdown_pending_ = true;
+  }
+}
+
+/******************************************************************************/
+void HandleMessage(const MessageHeader& header, const void* payload) {
+  if (header.message_type == MessageType::POSE) {
+    auto& contents = *static_cast<const PoseMessage*>(payload);
+
+    double p1_time_sec =
+        contents.p1_time.seconds + (contents.p1_time.fraction_ns * 1e-9);
+
+    static constexpr double SEC_PER_WEEK = 7 * 24 * 3600.0;
```

Comment:
this code is in 2 places. consider a static helper method or putting in defs or similar?

Reply:
Yeah, I'll move all of the message print stuff between this and the file decode example to a common function, similar to the Python examples.